lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -1585,6 +1585,13 @@ private static String validateName(String name) {
     return name;
   }
 
+  private static <T> T requireNotNull(T input, String name) {
```

Comment:
Maybe use `java.util.Objects#requireNonNull(T, java.lang.String)` ?

Reply:
this would throw an NPE and not an IllegalArgumentException. to most developers an NPE looks like a bug in the library rather than failed input validation.

but ok